src/bundle/EventListener/RequestAttributesListener.php

```diff
@@ -60,6 +67,18 @@ public function addRequestAttributes(FilterViewBuilderParametersEvent $event)
             $parameterBag->remove('locationId');
             $parameterBag->set('location', $location);
         }
+
+        if ($this->fetchingTranslatedContent($request, $parameterBag)) {
+            /** @var Location $location */
+            $location = $parameterBag->get('location');
+            $content = $this->loadContent($location->contentInfo->id, $parameterBag->get('language'));
+            $parameterBag->set('content', $content);
+        }
+    }
+
+    private function fetchingTranslatedContent(Request $request, ParameterBag $parameterBag)
+    {
+        return $parameterBag->has('language') && $parameterBag->has('location') && $request->get('_route') === '_ezpublishLocation';
```

Comment:
Wouldn't checking `_route` first be more logical, would it?

Reply:
`_route` check is the heaviest one